go-controller/pkg/node/management-port_linux.go

```diff
@@ -160,48 +160,34 @@ func tearDownInterfaceIPConfig(link netlink.Link, ipt4, ipt6 util.IPTablesHelper
 	return nil
 }
 
-func tearDownManagementPortConfig(mpcfg *managementPortConfig) error {
-	// for the initial setup we need to start from the clean slate, so flush
-	// all (non-LL) addresses on this link, routes through this link, and
-	// finally any IPtable rules for this link.
-	var ipt4, ipt6 util.IPTablesHelper
-
-	if mpcfg.ipv4 != nil {
-		ipt4 = mpcfg.ipv4.ipt
-	}
-	if mpcfg.ipv6 != nil {
-		ipt6 = mpcfg.ipv6.ipt
-	}
-	return tearDownInterfaceIPConfig(mpcfg.link, ipt4, ipt6)
-}
-
 func setupManagementPortIPFamilyConfig(routeManager *routemanager.Controller, mpcfg *managementPortConfig, cfg *managementPortIPFamilyConfig) ([]string, error) {
 	var warnings []string
 	var err error
 	var exists bool
 
-	if exists, err = util.LinkAddrExist(mpcfg.link, cfg.ifAddr); err == nil && !exists {
-		// we should log this so that one can debug as to why addresses are
-		// disappearing
-		warnings = append(warnings, fmt.Sprintf("missing IP address %s on the interface %s, adding it...",
-			cfg.ifAddr, mpcfg.ifName))
-		err = util.LinkAddrAdd(mpcfg.link, cfg.ifAddr, 0, 0, 0)
-	}
+	// synchronize IP addresses, removing undesired addresses
+	// should also remove routes specifying those undesired addresses
+	err = util.SyncAddresses(mpcfg.link, []*net.IPNet{cfg.ifAddr})
 	if err != nil {
 		return warnings, err
 	}
 
+	// now check for addition of any missing routes
 	for _, subnet := range cfg.allSubnets {
-		exists, err = util.LinkRouteExists(mpcfg.link, cfg.gwIP, subnet)
+		route, err := util.LinkRouteExists(mpcfg.link, cfg.gwIP, subnet)
 		if err != nil {
 			return warnings, err
 		}
-		if exists {
-			continue
+		// Check MTU manually since netlink doesn't support it as an RT_FILTER
+		if route != nil {
+			if route.MTU == config.Default.RoutableMTU {
+				continue
+			}
```

Comment:
Don't we need to add the route to the route manager anyway? Isn't it in charge of keeping routes in sync?

Reply:
good point...I was investigating this on an older version of code. In <=4.13 we will need to do something like this, but for later releases we can just rely on routeManager